DragaliaAPI/Features/Event/CombatEventController.cs

```diff
@@ -0,0 +1,92 @@
+﻿using DragaliaAPI.Controllers;
+using DragaliaAPI.Features.Reward;
+using DragaliaAPI.Features.Trade;
+using DragaliaAPI.Models.Generated;
+using DragaliaAPI.Services;
+using DragaliaAPI.Shared.MasterAsset;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DragaliaAPI.Features.Event;
+
+[Route("combat_event")]
+public class CombatEventController(
+    IUpdateDataService updateDataService,
+    IRewardService rewardService,
+    IEventService eventService,
+    ITradeService tradeService
+) : DragaliaControllerBase
+{
+    [HttpPost("get_event_data")]
+    public async Task<DragaliaResult> GetEventData(CombatEventGetEventDataRequest request)
+    {
+        CombatEventGetEventDataData resp = new();
+
+        resp.combat_event_user_data = await eventService.GetCombatEventUserData(request.event_id);
+        resp.event_reward_list = await eventService.GetEventRewardList<BuildEventRewardList>(
+            request.event_id
+        );
+        resp.user_event_location_reward_list =
+            await eventService.GetEventRewardList<UserEventLocationRewardList>(request.event_id);
+
+        int tradeGroupId = MasterAsset.EventTradeGroup.Enumerable
+            .First(x => x.EventId == request.event_id)
+            .Id;
```

Comment:
should this implement sorting? looks like there can be multiple with the same event_id

Reply:
no sorting but it should actually return all groups, fixed